lib/chef/knife/node_run_list_remove.rb

```diff
@@ -42,7 +42,18 @@ def run
           entries = @name_args[1].split(',').map { |e| e.strip }
         end
 
-        entries.each { |e| node.run_list.remove(e) }
+        # iterate over the list of things to remove,
+        # warning if one of them was not found
+        entries.each do |e|
+          if node.run_list.find { |rli| e == rli.to_s }
+            node.run_list.remove(e)
+          else
+            ui.warn "#{e} is not in the run list"
```

Comment:
When would we hit this? This error is very matter of fact and I think it would leave a user feeling, "ooookay?" We should include the run_list in the error in case it really is there but we're looking at the wrong thing (recipe::default versus recipe comes to mind).

Put another way, "Attempted adding #{e} to the run list, but an unknown error occurred and it wasn't found in the run list: #{node.run_list}"